### DISCUSSION TITLE
Filter data using the first character that user types in the input box

### DISCUSSION THREAD

**Discussion by developersatyajit** [JedWatson/react-select#4536] [Q&A] [answered]
I want to filter the data when user types the first character in the input box. Let say, the list is showing data like Andhrapradesh, Arunachal, Bihar, Boland, Bolivia, West Bengal, West Virginia. Now when user types only B, then Boland, Bolivia must be shown. Alternatively if he types W then West Bengal, West Virginia must be shown in the list. Right now I have to types 3-4 characters to get the expected result in the list. Is there any props available to do that?

**Comment by ebonow** [comment, marked as answer]
Greetings @developersatyajit 

It sounds like you would want to create a filter to change the logic being used for string matching. There is some documentation [here](https://react-select.com/advanced#custom-filter-logic), and it looks like you want to use `createFilter` to change the default value of `matchFrom` to `start` instead of `any`

```jsx
import { createFilter }, Select from 'react-select';

const filterOption = createFilter({ matchFrom: 'start' });

return <Select filterOption={filterOption} {...otherProps} />
```

Please let me know if this works for you.

**Reply by developersatyajit** [reply]
Thanks Ebonow. You have saved my day. Its working just like I expected and very easy to use. Thanks again
